beep/utils/waveform.py

```diff
@@ -150,11 +155,11 @@ def get_input_currents_both_to_final_soc(self, charging_c_rates):
                                maxfev=1000,
                                factor=0.1,
                                epsfcn=0.01,
-                               args=(end_soc_time, charging_c_rates))
+                               args=(end_soc_time, charging_c_rates, mesh_points))
 
         charging_c_rates = list(np.add(charging_c_rates, offset_solved))
```

Comment:
Change line to
charging_c_rates = [charging_c_rates[0]] + list(np.add(charging_c_rates[1:-1],offset)) + [charging_c_rates[-1]]
This change makes it so that only the non-edge c_rate values are offset.

Reply:
changed this one too